### ISSUE TITLE
login

### ISSUE THREAD

**Issue by hauweg** [fetlife/android#304]
It doesn't keep me logged in... Everytime I open the app I need to login again :-( 
Samsung, S4

**Comment by FernandoMiguel** [comment]
that has been fixed in the latest version.
please upgrade
https://github.com/fetlife/android/releases

**Comment by wolfxlittlered** [comment]
V2.6.19 I have actually been logged out when switching from mobile data to WiFi and vice versa. Have yet to duplicate it. However there is an open issue of this titled "frequent logins".

**Comment by kistitan** [comment]
Hi @wolfxlittlered,

Thank you for reporting this issue.
If you use 2.6.19 than it's a new issue as the previous one has been resolved.
May I ask if it is continuously reproducible?

Thanks

**Comment by LanceLake** [comment]
When I download any apk file, the Android system says I can not open it.

**Comment by FernandoMiguel** [comment]
@LanceLake have you enabled untrusted sources?
http://minotaurproject.co.uk/Minotaur/androidinstall.php

**Comment by LanceLake** [comment]
Yes. The apk files seem to be corrupted when downloaded. I tried the
versions before the released one.

**Comment by kistitan** [comment]
Hmm...
It is a strange issue
APK file should be ok as it can be installed on other devices.
Can you open from s different App? Like downloads, file browser?

**Comment by LanceLake** [comment]
I can download it. But when I try to run it on my phone, I get an error.
.05 works fine.

**Comment by kistitan** [comment]
Can you maybe delete the file and download again?
I just tested and install started without any issues.

**Comment by LanceLake** [comment]
Already tried. No luck. If it helps, I'm using 6.01

**Comment by LanceLake** [comment]
https://goo.gl/photos/5ruVUHB62GEnpPmA7

**Comment by LanceLake** [comment]
I see the issue.

https://stackoverflow.com/questions/3253079/cant-install-apk-from-browser-downloads

For everyone who is having this issue, download it and use the file
manager. Works for me now. Apparently, Chrome started blocking running APKs
within the browser.

**Comment by wolfxlittlered** [comment]
@Kistitan. I have not been able to duplicate the issue. However the fetlife app was the only one to cycle me out when switching. So far it has not happened since. It was a public WiFi as well. Not sure if maybe it was considered an unknown source when connecting to WiFi. It also cycled me out when I switched back to mobile data from WiFi. Again, unable to duplicate on v2.6.19

**Comment by kistitan** [comment]
Hi @wolfxlittlered  and @LanceLake 

Thank you for your comments. 
Apk file still seems to be valid, but your comments should and findings should go into our troubleshooting guide.

Thank you.

**Comment by MaxiM-PL** [comment]
Login issue is back with a vengeance in the latest preview. Galaxy S4 as well.

**Comment by kistitan** [comment]
Hi @MaxiM-PL

Login issue means you have to keep signing in even when you checked the remember me box?

I cannot reproduce it with 2.6.22.

Can you please confirm it is this issue and version?

Thanks

**Comment by MaxiM-PL** [comment]
Yup. Upgraded to 2.6.22 few hours ago and now I have to log in every time I launch the app.

**Comment by kistitan** [comment]
And just for the record: remember be check box is checked during login...

Hmm.

**Comment by MaxiM-PL** [comment]
Quick update: clearing cache and data seemingly helped. Tried killing and relaunching app few times and seems to be working for now. Possible false alarm for whatever reason after upgrading.

**Comment by kistitan** [comment]
Thank you for letting us know.
It can still be an issue in the App.
I will test upgrade from several versions.

Do you remember what version did you use before upgrade?
Do you use only one FetLife account?

Thank you

**Comment by MaxiM-PL** [comment]
I keep my app up to date as much as possible so probably I was using whatever version was latest. And yeah, I'm only using one account.

**Comment by wolfxlittlered** [comment]
V2.6.24. Issue still present. "Connection unavailable" message. Asked to log in every time after closing the app. Connection is secure and continuios so unsure how I am receiving the error message.

**Comment by wolfxlittlered** [comment]
One more time. V2.6.25. Same issue as comment above.

**Comment by kistitan** [comment]
@wolfxlittlered 

We cannot reproduce this issue on any of our test devices.
If we build a specific test version with extended login. Would you mind to install it to help us out?

Thanks and sorry for the inconveniences.

**Comment by steelowheel** [comment]
The app keeps me logged in, but every time I update the app, it logs me out. This normal? I know this never happened a month or two ago.

**Comment by wolfxlittlered** [comment]
I would be glad to contribute any way I can.

**Comment by kistitan** [comment]
@steelowheel

Not every update logs you out, but recent update we indeed made it that way you have to log in again due to changes in the user session handling.

Sorry for the inconveniences

**Comment by wolfxlittlered** [comment]
Android 5.1.1
Kernel 3.10.49
Not sure if my phone software makes a difference.

**Comment by kistitan** [comment]
Thank you @wolfxlittlered 

Can you please rename the attached file to .apk, than install it as usual?
After install please login as usual and than open the menu and press on the new "send logs" menu item.
Please reproduce the issue a few times sending the logs after each login.

Thanks a lot, your help is really appreciated.
 
[fetlife-2.6.27.zip](https://github.com/fetlife/android/files/990427/fetlife-2.6.27.zip)

**Comment by wolfxlittlered** [comment]
@kistitan it says "cannot parse package" do I need any additional apps or files?

**Comment by kistitan** [comment]
@wolfxlittlered 

Hmm... maybe it is because of the upload as zip.
Can you please write me an email to Krisztian@fetlife.com?

I will send the the file there to you.

Thanks

**Comment by wolfxlittlered** [comment]
@kistitan Done. Used analternate email but the subject line is "Github wolfxlittlered".

**Comment by MaxiM-PL** [comment]
Hello again, guess what? ;] 
Now it's different though: first I'm getting some error message about being disconnected then I have to log in again. Latest release.

**Comment by wdomdm** [comment]
I am on a Motorola Droid Turbo 2. Yesterday I downloaded the Official FetLife App v2.6.26.  Everytime I get a notification and click on it, it takes me to the correct place for 1 second then redirects to the login screen. Blank login screen. 
I have checked the settings to make sure that I do not have the box checked to log in each time. I do not. 
Log in and check the box to remember me. 

Previous versions did it occasionally. This is every single time. Please fix asap. Thank you very much.

**Comment by kistitan** [comment]
Hi there,

AL you have frequent login issues, can you please try

https://github.com/fetlife/android/releases/download/v2.6.27/fetlife-2.7.0.apk

If it does not work, clearing the data should do the trick.

Thank you and sorry for the lots of inconveniences.
We are having hard time with a library we use.

Thanks again

**Comment by kistitan** [comment]
Hi there,

Anybody has any experience with 2.7.0?
Is it still happening?

**Comment by MaxiM-PL** [comment]
@kistitan My issue seems to be gone in 2.7, thanks.

**Comment by kistitan** [comment]
Please let me know if anybody still facing the issue with the new version

**Comment by wolfxlittlered** [comment]
I'm having trouble downloading. I'm sure its on my end.

**Comment by wolfxlittlered** [comment]
The issue was available space. Android 5.1.1 does not give insufficient storage message. Make sure you have available space for the download.

**Comment by wolfxlittlered** [comment]
So far so good. Still logged in even after switching data to WiFi and back.

**Comment by VaanAuron** [comment]
Yes I have had the same problem with it logging me out when ever I put the app in background for  like last 3 updates. I just installed the 2.7.0 and now it not logging me out.

**Comment by nextgen-networks** [comment]
Login issue is here again since Friday May 19th.
I use app version 2.7.0.

**Comment by xeroace** [comment]
Having to login after I close out the app. Currently on version 2.7.0 using a galaxy 8

**Comment by kistitan** [comment]
Hi there,

This issue is very annoying for long. Both on user and on developer site.
It really seems we have reached the limit of the usage of our db library and need some structural changes.

I will try to find a quick workaround first before starting the deeper and probably more time consuming restructure.

**Comment by kdom13** [comment]
I'm having this issue too, on 2.6.24.  And it also doesn't keep my info cached in the app so I can just click login, at least.  I have to type it in each time.

side note - I don't see a download option for 2.7

**Comment by kistitan** [comment]
For all those who are facing issues. Can you please clear all storage data in 
Devife settings / Apps / Fetlife?
Please let me Kno if it helps.

Thank you

**Comment by nextgen-networks** [comment]
@kistitan cleaning app cache and app-data fixed the login-problem with my installed 2.7.0 version

**Comment by kdom13** [comment]
I upgraded and cleared cache at the same time, so I'm not sure which fixed it, but as of now 2.7.0 is working here as well.

**Comment by wolfxlittlered** [comment]
Clearing cache did not resolve issue. V2.7.0 android v5.1.1

**Comment by kistitan** [comment]
Thank you for letting know.
I am afraid it will be a bigger change them.
I am sorry.
Still looking for a quick temporary solution.
I hope tomorrow I can release one.

**Comment by kistitan** [comment]
Issue seems to be solved, we got report for frequent login issues only on one device with a strange brand.
If anybody sees the issue, please let us know.

note:
- first please be sure you cleared all data of the app
- rare forced login is normal as for security reasons tokens get expired

**Comment by Two-Finger** [comment]
The issue started happening to me too, but only recently, probably after the upgrade to 2.7
Once I stop the app (by swiping it away in Recent Apps, or force-stop it in Application manager​), next time I open it, it flashes a few times and says "Please login again for security reasons".

Phone: Samsung Galaxy S4 Mini Plus (I9195I)
Android: 4.4.4
